client/components/translatable/proptype.js

```diff
@@ -0,0 +1,62 @@
+/** @format */
+
+function translatableStringChecker( props, propName, componentName ) {
```

Comment:
It is worth export this func and writing a test?

Reply:
Definitely. I'd be grateful for a little guidance! So how would I best approach this?

Reply:
I was half-way through a post, then I just though it would be easier to create a PR for this branch: #22031

While testing I noticed that we're letting through nil values (`null`, `undefined`) and `0`, while other numbers `2` will be validated. 

The tests I wrote are there to be tweaked :) It depends on how strict we want to be.

Reply:
I've updated code and tests to not allow numbers.